adapters/Unity/UnityAdapter/GADMAdapterUnity.m

```diff
@@ -81,17 +80,25 @@ - (void)getInterstitial {
     [strongConnector adapter:self didFailAd:error];
     return;
   }
-  _isLoading = YES;
-  [[GADMAdapterUnitySingleton sharedInstance] requestInterstitialAdWithDelegate:self];
+
+  UnitySingletonCompletion completeBlock = ^(UnityAdsError *error, NSString *message) {
+    if(!error) {
+            [UnityAds addDelegate:self];
+            [UnityAds load:[self getPlacementID]];
+    } else {
+      if (strongConnector) {
+        NSError *errorWithDescription = GADUnityErrorWithDescription(message);
+        [strongConnector adapter:self didFailAd:errorWithDescription];
+      }
+    }
```

Comment:
Can be simplified to avoid nesting? Apply through out all files.

```
if (error) {
  if (strongConnector) {
    NSError *errorWithDescription = GADUnityErrorWithDescription(message);
    [strongConnector adapter:self didFailAd:errorWithDescription];
  }
  return;
}

[UnityAds addDelegate:self];
[UnityAds load:_placementID];
```